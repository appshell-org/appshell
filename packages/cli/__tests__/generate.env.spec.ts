import fs from 'fs';
import path from 'path';
import { rimrafSync } from 'rimraf';
import generateEnvHandler from '../src/handlers/generate.env';

describe('generate.env', () => {
  const prefix = '';
  const outDir = path.resolve('packages/cli/__tests__/assets/temp');
  const outFile = 'appshell.env.js';
  const globalName = 'appshell_env';

  beforeAll(() => {
    process.env.REGISTRY = 'packages/cli/__tests__/assets/appshell_registry';
    process.env.ROOT = 'TestModule/Workspace';
    process.env.TEST_ENV_FOO = 'foo';
    process.env.TEST_ENV_BAR = 'bar';
  });

  afterAll(() => {
    delete process.env.REGISTRY;
    delete process.env.ROOT;
    delete process.env.TEST_ENV_FOO;
    delete process.env.TEST_ENV_BAR;
  });

  afterEach(() => {
    rimrafSync(outDir);
  });

  it('should generate the runtime environment js file', async () => {
    await generateEnvHandler({
      prefix: '^(TEST_|REGISTRY|ROOT).*',
      outDir,
      outFile,
      globalName,
    });
    const actual = fs.readFileSync(path.join(outDir, outFile));

    expect(actual.toString()).toMatchSnapshot();
  });

  it('should capture only prefixed environment vars when prefix is supplied', async () => {
    await generateEnvHandler({
      prefix: 'TEST_',
      outDir,
      outFile,
      globalName,
    });
    const actual = fs.readFileSync(path.join(outDir, outFile));

    expect(actual.toString()).toMatchSnapshot();
  });

  it('should output to outDir when supplied', async () => {
    const testPath = path.resolve('packages/cli/__tests__/assets/temp/test');
    await generateEnvHandler({
      prefix,
      outDir: testPath,
      outFile,
      globalName,
    });
    const actual = fs.readFileSync(path.join(testPath, outFile));

    expect(actual.toString()).toBeTruthy();
  });

  it('should output with outFile name when supplied', async () => {
    const filename = 'test.env.js';
    await generateEnvHandler({
      prefix,
      outDir,
      outFile: filename,
      globalName,
    });
    const actual = fs.readFileSync(path.join(outDir, filename));

    expect(actual.toString()).toBeTruthy();
  });

  it('should output with global variable name supplied', async () => {
    const testGlobalName = 'my_global_var';
    await generateEnvHandler({
      prefix,
      outDir,
      outFile,
      globalName: testGlobalName,
    });
    const actual = fs.readFileSync(path.join(outDir, outFile));

    expect(actual.includes(`window.${testGlobalName}`)).toBe(true);
  });
});
