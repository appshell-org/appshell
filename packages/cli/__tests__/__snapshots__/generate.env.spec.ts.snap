// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generate.env should capture only prefixed environment vars when prefix is supplied 1`] = `
"window.appshell_env = {
	TEST_ENV_FOO: 'foo',
	TEST_ENV_BAR: 'bar',
}"
`;

exports[`generate.env should generate the runtime environment js file 1`] = `
"window.appshell_env = {
	REGISTRY: 'packages/cli/__tests__/assets/appshell_registry',
	ROOT: 'TestModule/Workspace',
	TEST_ENV_FOO: 'foo',
	TEST_ENV_BAR: 'bar',
}"
`;
